src/commands/moderators/stickyMessage.ts

```diff
@@ -0,0 +1,146 @@
+import {
+  ApplicationCommandOptionType,
+  ApplicationCommandType,
+  PermissionsBitField,
+  TextChannel,
+} from 'discord.js'
+
+import { prisma } from '../../prisma.js'
+import { defineCommandHandler } from '../../types/defineCommandHandler.js'
+import { saveCache } from '../../utils/cache.js'
+
+export default [
+  defineCommandHandler({
+    data: {
+      name: 'stickao-set',
+      description:
+        'set sticky message in current channel (will reset every 5 message)',
+      defaultMemberPermissions: PermissionsBitField.Flags.ManageMessages,
+      type: ApplicationCommandType.ChatInput,
+      options: [
+        {
+          name: 'message',
+          description:
+            'The message that you want to sticky in the bottom of channel (required)',
```

Comment:
question - is it possible to create message with multiline when using a slash command?

example sticky message in รายงานสแปม channel:

<img width="777" alt="image" src="https://github.com/creatorsgarten/kaogeek-discord-bot/assets/193136/f962e8ee-7526-4137-a5c4-e8c8a67435f2">

```
# ห้องนี้มีไว้สำหรับแจ้งผู้กระทำผิดเท่านั้นนะครับ งดคุยเรื่องอื่น ๆ 

สำหรับผู้แจ้ง โปรดบอกชื่อ ภาพลิงก์ข้อความที่ชี้ไปยังข้อความที่กระทำผิด พร้อม Tag Moderator  หรือ Admin
สำหรับห้อง Voice โปรดแนบ Video ที่เป็นหลักฐานเพื่อให้ตัดสินใจได้ง่านขึ้น

สำหรับการแจ้งโดยไม่มีหลักฐานใด ๆ ขอไม่รับเคสนะครับ

ขอบคุณครับ
```

Reply:
I seem like need to done with modal like @Leomotors said

https://github.com/discord/discord-api-docs/issues/2381

Reply:
We are still having problem with modal, per [Issue 100](https://github.com/creatorsgarten/kaogeek-discord-bot/issues/100). I am working on a way to fix without skipping `deferReply` logic. Might need to push modal to todo first.

Reply:
In my way, I fix this by not setting deferReply by default but setting it on top of the command that wants defer.

```typescript
// src\events\interactionCreate.ts
import { Events } from 'discord.js'

import { defineEventHandler } from '../types/defineEventHandler.js'

export default defineEventHandler({
  eventName: Events.InteractionCreate,
  once: false,
  execute: async (botContext, interaction) => {
    if (interaction.isCommand()) {
      const commandName = interaction.commandName
      const { commands } = botContext
      const command = commands.get(commandName)
      if (!command) return
      try {
        console.log(
          `[Command] ${interaction.user.tag} (${interaction.user.id}) > ${interaction.commandName}`,
        )
        await command.execute(botContext, interaction)
      } catch (error) {
        console.error(error)
      }
    }
  },
})
``` 
e.g. ping command
```typescript
import { defineCommandHandler } from '../../types/defineCommandHandler.js'

export default defineCommandHandler({
  data: {
    name: 'ping',
    description: 'Ping!',
  },
  execute: async ({ client }, interaction) => {
    await interaction.deferReply({ ephemeral: true })

    await interaction.editReply({
      embeds: [
        { description: `ความหน่วง! ${client.ws.ping}ms`, color: 0x00ff00 },
      ],
    })
  },
})
``` 
Does anyone have any other solutions?